qp_shotgun/humann2/humann2.py

```diff
@@ -0,0 +1,132 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import basename, join
+
+from future.utils import viewitems
+import pandas as pd
+
+
+def generate_humann2_analysis_commands(forward_seqs, reverse_seqs, map_file,
+                                       out_dir, parameters):
+    """Generates the humann2 commands
+
+    Parameters
+    ----------
+    forward_seqs : list of str
+        The list of forward seqs filepaths
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    map_file : str
+        The path to the mapping file
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+
+    Returns
+    -------
+    list of str
+        The humann2 commands
+
+    Raises
+    ------
+    ValueError
+        If the rev is not an empty list and the same length than fwd seqs
+        The prefixes of the run_prefix don't match the file names
+
+    Notes
+    -----
+    The forward and reverse files are going to have different filenames but
+    the sample name is going to be the same so the results are merged when
+    joining the outputs
+    """
+    # making sure the forward and reverse reads are in the same order
+    forward_seqs.sort()
+    if reverse_seqs:
+        if len(forward_seqs) != len(reverse_seqs):
+            raise ValueError('Your reverse and forward files are of different '
+                             'length. Forward: %s. Reverse: %s.' %
+                             (', '.join(forward_seqs),
+                              ', '.join(reverse_seqs)))
+        reverse_seqs.sort()
+
+    sn_by_rp = get_sample_names_by_run_prefix(map_file)
+
+    # we match sample name and forward filename
+    samples = []
+    for i, fname in enumerate(forward_seqs):
+        f = basename(fname)
+        # removing extentions: fastq or fastq.gz
+        if 'fastq' in f.lower().rsplit('.', 2):
+            f = f[:f.lower().rindex('.fastq')]
+        # this try/except block is simply to retrieve all possible errors
+        # and display them in the next if block
+        try:
```

Comment:
Sounds good.

On (Sep-04-16|10:00), Antonio Gonzalez wrote:
> > +                             (', '.join(forward_seqs),
> > +                              ', '.join(reverse_seqs)))
> > +        reverse_seqs.sort()
> > +
> > +    sn_by_rp = get_sample_names_by_run_prefix(map_file)
> > +
> > +    # we match sample name and forward filename
> > +    samples = []
> > +    for i, fname in enumerate(forward_seqs):
> > +        f = basename(fname)
> > +        # removing extentions: fastq or fastq.gz
> > +        if 'fastq' in f.lower().rsplit('.', 2):
> > +            f = f[:f.lower().rindex('.fastq')]
> > +        # this try/except block is simply to retrieve all possible errors
> > +        # and display them in the next if block
> > +        try:
> 
> The issue is that that block can fail in any of those lines and the result should be the same: pass. Thus, it will become 2 try/except clauses. 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/qiita-spots/qp-shotgun/pull/2/files/f61ecb3cfc94a51fb45884d4d710334284e24627#r77455635